From: Michael Tokarev <mjt@tls.msk.ru>
Subject: openbios: spelling: endiannes
Forwarded: no

diff --git a/roms/openbios/kernel/bootstrap.c b/roms/openbios/kernel/bootstrap.c
index b7658ab6e9..fb4506e3ac 100644
--- a/roms/openbios/kernel/bootstrap.c
+++ b/roms/openbios/kernel/bootstrap.c
@@ -173,5 +173,5 @@ static void write_dictionary(const char *filename)
 #ifdef CONFIG_BIG_ENDIAN
-		.endianess	= -1,
+		.endianness	= -1,
 #else
-		.endianess	= 0,
+		.endianness	= 0,
 #endif
diff --git a/roms/openbios/kernel/dict.c b/roms/openbios/kernel/dict.c
index 0986cb14f0..51afc6775f 100644
--- a/roms/openbios/kernel/dict.c
+++ b/roms/openbios/kernel/dict.c
@@ -254,3 +254,3 @@ void dump_header(dictionary_header_t *header)
 	printk("  cellsize:    %d\n", header->cellsize);
-	printk("  endianess:   %s\n", header->endianess?"big":"little");
+	printk("  endianness:  %s\n", header->endianness?"big":"little");
 	printk("  compression: %s\n", header->compression?"yes":"no");
diff --git a/roms/openbios/kernel/include/dict.h b/roms/openbios/kernel/include/dict.h
index 749fd6fbaf..1e735fa09b 100644
--- a/roms/openbios/kernel/include/dict.h
+++ b/roms/openbios/kernel/include/dict.h
@@ -29,3 +29,3 @@ typedef struct dictionary_header {
 	u8	cellsize;
-	u8 	endianess;
+	u8 	endianness;
 	u8	compression;
